### DISCUSSION TITLE
can you create a completely anonymous sandbox?

### DISCUSSION THREAD

**Discussion by danielkrajnik** [netblue30/firejail#4819] [Q&A] [answered]
Is there a way to start firejail, so that it creates a sandbox as anonymous as possible? Running programs in VMs had this advantage that even if a program "called home" it had a harder time to tell who run it.

`--private` flag accopmlishes this for home directory I think, but not for `/` root (e.g. `/secret-dir` still shows up)

**Comment by rusty-snake** [comment, marked as answer]
> `--private` flag accopmlishes this for home directory I think, but not for `/` root

`man 1 firejail`:

> **--chroot=dirname**
> Chroot the sandbox into a root filesystem.

**Reply by danielkrajnik** [reply]
thank you again for the fast response. Wow, `--chroot` mode starts to remind me of containers. Perfect, now it makes sense

**Comment by ghost** [comment]
There might be some confusion here. When I read the question, my first thought was it refered to topics like collecting telemetry data, IP address, tracking data etcetera. So things relating to `anonimity/privacy`, that need `networking activity` to deliver that kind of data back to "the mothership" (a.k.a. "phoning home"). To me that's not quite the same as trying to block access to specific parts of the filesystem (like user/root's home directories). And if that's the gist of your question things get a bit more complex. Firejail != Tor, but there are several options that are relevant in this context.

`man 1 firejail`:

--dns=address
    Set a DNS server for the sandbox. Up to three DNS servers can be defined. Use this option if you don't trust the DNS setup on your network.

--hosts-file=file
    Use file as /etc/hosts.

--net=none
    Enable a new, unconnected network namespace. The only interface available in the new namespace is a new loopback interface (lo). Use this option to deny network access to programs that don't really need network access.

--netfilter
    Enable a default firewall if a new network namespace is created inside the sandbox.

But, as I said, I'm not sure my reading of the question is correct. If not, please disregard :-)

**Reply by rusty-snake** [reply]
tor(not browser)+firejail:
- https://github.com/equk/torjail
- https://github.com/orjail/orjail
